data/campaigns/Northern_Rebirth/scenarios/07a_Settling_Disputes.cfg

```diff
@@ -158,51 +160,51 @@
             {TRAIT_HEALTHY}
         [/modifications]
 
-        hitpoints=1
+        {QUANTITY hitpoints 3 2 1}
         fog=yes
         share_vision=all
         facing=se
 
         # Place some combatants at various states of hp/xp
         {NOTRAIT_UNIT 2 "Dwarvish Ulfserker" 28 8}
         [+unit]
-            hitpoints=12
-            experience=40
+            {QUANTITY hitpoints 16 14 12}
+            {QUANTITY experience 41 40 39}
```

Comment:
this makes him weaker on HARD

Reply:
I'll have to review this one once I'm back on my computer (on mobile currently)...

Reply:
So, right, an Ulfserker's max experience is 42. I can't go any higher without making him auto-advance, and I generally like to keep all 3 numbers different from one another, in a linear pattern...

Reply:
@cooljeanius
> I generally like to keep all 3 numbers different from one another, in a linear pattern

why? There's no need to differentiate the numbers for the sake of just being different, especially if the difference is this minuscule. It makes both maintaining a scenario a chore (more values to change on refactors etc), and, in this and many similar places, serves no actual purpose. I'd mirror what @stevecotton said before: try to find an aim first, and propose the changes for that next.

If the unit is about to get AMLAd anyway, the HPs doesn't matter as long as it's the same amount of troll hits. Same with XP: if the unit is fighting trolls, then 40 XP and 41 XP are effectively the same, they'll either get AMLAd or killed, no other option here.

If you want to make the units able to survive more hits, you'll need to give them more than 2 HPs in this case (AFAIR, troll hits for 10-ish dam). Tweaking XP makes no sense whatsoever, with 40 XP it's just a single fight. That's why you're breaking HARD here, because 39 XP means surviving two troll combats, so it's virtually impossible and you're marking this unit to die.

Reply:
> @cooljeanius
> 
> > I generally like to keep all 3 numbers different from one another, in a linear pattern
> 
> why? There's no need to differentiate the numbers for the sake of just being different, especially if the difference is this minuscule.

Maybe not a NEED, but it's still a want, at least.

> It makes both maintaining a scenario a chore (more values to change on refactors etc), and, in this and many similar places, serves no actual purpose.

It increases replayability, though.

> I'd mirror what @stevecotton said before: balancing the scenario is not about having aesthetic numbers. 

I disagree, aesthetic numbers are nice.

> It's about trying to find an aim first, and proposing the changes for that next.
> 
> If the unit is about to get AMLAd anyway, the HPs doesn't matter as long as it's the same amount of troll hits. Same with XP: if the unit is fighting trolls, then 40 XP and 41 XP are effectively the same, they'll either get AMLAd or killed, no other option here.
> 
> If you want to make the units able to survive more hits, you'll need to give them more than 2 HPs in this case (AFAIR, troll hits for 10-ish dam, both 12, 14 and 16 are two hits). Tweaking XP makes no sense whatsoever, with 40 XP it's just a single fight. That's why you're breaking HARD here, because 39 XP means surviving two troll combats, so it's virtually impossible and you're marking this unit to die.

Well anyways, I'm no longer breaking HARD as of [0149764](https://github.com/wesnoth/wesnoth/pull/6057/commits/014976436016a4f09260a5e12fecc066999ed833) (untested).

Reply:
You do realize that `{QUANTITY 41 40 40}` works just fine, right? Also, does 18 represent more hits than 12?

Reply:
> You do realize that `{QUANTITY 41 40 40}` works just fine, right?

Yeah but I don't like repeating numbers between difficulties; I prefer the ifdef method in that case